lib/api/controllers/routerController.js

```diff
@@ -418,15 +215,44 @@ module.exports = function RouterController (kuzzle) {
     api.get('/', (request, response) => {
       var _routes = {};
 
-      extractRouteInfo(_routes, routes);
-      extractRouteInfo(_routes, kuzzle.pluginsManager.routes, '/_plugin');
+      extractRouteInfo(_routes, routes, '/' + kuzzle.config.apiVersion);
+      extractRouteInfo(_routes, [{verb: 'get', url: '/_serverInfo', controller: 'read', action: 'serverInfo'}]);
+      extractRouteInfo(_routes, kuzzle.pluginsManager.routes, '/' + kuzzle.config.apiVersion + '/_plugin');
 
       response.setHeader('Access-Control-Allow-Origin', '*');
       response.setHeader('Access-Control-Allow-Headers', 'X-Requested-With');
       response.writeHead(200, {'Content-Type': 'application/json'});
       response.end(stringify({status: 200, error: null, result: {message: 'Available routes for this API version by verb.', routes: _routes}}));
     });
 
+    apiBase.get('/swagger.json', (request, response) => {
+      var content;
+      try {
+        content = fs.readFileSync(__dirname + '/../../../kuzzle-swagger.json');
+      }
+      catch (e) {
+        content = stringify({error: 'The swagger file has not been generated. Use the `kuzzle swagger` CLI command to generate it.', status: 500});
```

Comment:
The solution I propose wouldn't generate swagger files. What I propose is to generate *swagger documentation* dynamically and, if called by the CLI, to store this documentation into files. This would answer to the points you stated.

The problem I see with your solution is a practical one: like you said, swagger routes should be used scarcely. But when it's needed, the probability that the files haven't been generated, or that they are outdated, is very high. 
And, of course, there is the problem of having an enabled HTTP route basically answering that an administrator action is needed first. That's not very practical either.

Here is an example: let's assume that you install Kuzzle and generate those files. Then you run it for a long time, patch it several times and also install a bunch of controller plugins, without ever generating the swagger files. When you suddenly need the documentation using HTTP, you get a dangerously outdated one. And you don't even know it.

One last matter: I didn't see where you inject controller plugins swagger documentation in the files. Can you point me to it?